splitio/client/factory.py

```diff
@@ -684,7 +696,16 @@ def get_factory(api_key, **kwargs):
     _INSTANTIATED_FACTORIES.update([api_key])
     _INSTANTIATED_FACTORIES_LOCK.release()
 
-    config = sanitize_config(api_key, kwargs.get('config', {}))
+    config_raw = kwargs.get('config', {})
```

Comment:
I am not sure if we need the global variables _TOTAL_FLAG_SETS, _INVALID_FLAG_SETS and _FLAG_SETS_LOCK to calculate total_flag_sets and invalid_flag_sets, because they are calculated per factory instance. Look if the following can work and help to simplify the code:
- remove lines 700 to 706
- replace `= _get_total_and_invalid_flag_sets()` with `= _get_total_and_invalid_flag_sets(cfg)`
- and implement `_get_total_and_invalid_flag_sets` as follows:
```
// Assuming that cfg has the same interface than config_raw
def _get_total_and_invalid_flag_sets(cfg):
    total_flag_sets = len(cfg.get('flagSetsFilter'))
    invalid_flag_sets = total_flag_sets - len(input_validator.validate_flag_sets(cfg.get('flagSetsFilter'), 'Telemetry Init'))
    return total_flag_sets, invalid_flag_sets
```

Reply:
Hi Emi, Every new factory instance will use the same static functions to create a factory instance, so the flag set calculation code could be  running for multiple factories at the same time. 
Its the same mechanism for the factories count per api key.